samples/dotnet/kernel-syntax-examples/Reliability/RetryThreeTimesWithRetryAfterBackoff.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Reliability;
+using Polly;
+using Polly.Retry;
+
+namespace Reliability;
+
+/// <summary>
+/// An example of a retry mechanism that retries three times with backoff using the RetryAfter value.
+/// </summary>
+public class RetryThreeTimesWithRetryAfterBackoff : IHttpRetryPolicy
+{
+    public Task<HttpResponseMessage> ExecuteWithRetryAsync(Func<Task<HttpResponseMessage>> request, ILogger log, CancellationToken cancellationToken = default)
+    {
+        var policy = GetPolicy(log);
+        return policy.ExecuteAsync((_) => request(), cancellationToken);
+    }
+
+    private static AsyncRetryPolicy<HttpResponseMessage> GetPolicy(ILogger log)
+    {
+        return Policy
+            .HandleResult<HttpResponseMessage>(response =>
+                response.StatusCode is System.Net.HttpStatusCode.TooManyRequests or System.Net.HttpStatusCode.Unauthorized)
```

Comment:
Have never seen retries for 401s.

Reply:
Yeah, it's just an example to demonstrate the functionality and is easy to reproduce when using an invalid key. Will leave some comments to make this more clear.